GVFS/GVFS.UnitTests/Mock/Common/MockLocalGVFSConfigBuilder.cs

```diff
@@ -0,0 +1,57 @@
+using GVFS.Common;
+using System.Collections.Generic;
+
+namespace GVFS.UnitTests.Mock.Common
+{
+    public class MockLocalGVFSConfigBuilder
```

Comment:
I like this pattern, there are a couple of things that I would do differently.
1. set the default values from the test here in the constructor and put them in the `entries` since that is the default case happy path good config.
2. Do not use default values in your `With...` methods because I don't know from looking at the test what the value of it will be.
3. You can use `WithNo...` to remove the default value from the entries.

I think this will make it more explicit what you are testing so the name of the test matches what your builder is calling. 

Like this
```
        public void NoUpgraderWhenNuGetFeedMissing()
        {
            MockLocalGVFSConfig gvfsConfig = this.ConstructMockLocalGVFSConfigBuilder()
                .WithNoUpgradeFeedUrl()
                .Build();
```
instead of 
```
        public void NoUpgraderWhenNuGetFeedMissing()
        {
            MockLocalGVFSConfig gvfsConfig = this.ConstructMockLocalGVFSConfigBuilder()
                .WithUpgradeRing()
                .WithUpgradeFeedPackageName()
                .Build();
```

Reply:
Thanks!

I incorporated some of these ideas. In the test setup, I create the "default" / "baseline" config for NuGet or GitHub, and then in the specific test cases I remove entries as appropriate with the `WithNo...` pattern.